### PR TITLE
Embedded js: remove more dependency on jquery

### PR DESCRIPTION
**Pull request by stsewd**
This is on top of https://github.com/readthedocs/readthedocs.org/pull/9508

---
:books: Documentation previews :books:

- User's documentation (`docs`): https://docs--9515.org.readthedocs.build/en/9515/

- Developer's documentation (`dev`): https://dev--9515.org.readthedocs.build/en/9515/

### CODE REVIEW

File: readthedocs/core/static-src/core/js/doc-embed/search.js [hunk 1]
```diff
@@ -2,28 +2,11 @@
...
-        for (let attr of Object.keys(attributes)) {
-            node.setAttribute(attr, attributes[attr]);
-        }
-    }
-    return node;
-};
-
+const rtddata = require('./rtd-data');
+const xss = require('xss/lib/index');
+const { createDomNode } = require("./utils");
```
Comment (ericholscher) [inline readthedocs/core/static-src/core/js/doc-embed/search.js:7]: Do we need to specify that we're using this function from `utils`, or is the compiler smart enough to know what is happening if we add additional utils functions?

Comment (agjohnson) [reply readthedocs/core/static-src/core/js/doc-embed/search.js:7]: The compile step will resolve the functions in `utils.js` for use here, yeah. This syntax will only import `createDomNode` from `utils.js`. If more functions are added to `utils.js`, they won't automatically be imported here, but could be with `const { createDomNode, someOtherUtil } = ..`

File: readthedocs/core/static-src/core/js/doc-embed/utils.js [hunk 2]
```diff
@@ -0,0 +1,18 @@
+/**
+ * Create and return DOM nodes with given attributes.
+ *
+ * @param {String} nodeName name of the node
+ * @param {Object} attributes obj of attributes to be assigned to the node
+ * @return {Object} DOM node
```
Comment (agjohnson) [inline readthedocs/core/static-src/core/js/doc-embed/utils.js:6]: Just a note for the future: your syntax is valid jsdoc tag syntax, but adding a hyphen makes it a bit easier for humans to scan/read:

```suggestion
 * @param {String} nodeName - name of the node
 * @param {Object} attributes - obj of attributes to be assigned to the node
 * @return {Object} DOM node
```

**Review by agjohnson — APPROVED** [review]

File: readthedocs/core/static-src/core/js/doc-embed/utils.js (hunk 2 above)
Comment (humitos) [reply readthedocs/core/static-src/core/js/doc-embed/utils.js:6]: Doesn't work with `:` as well? It looks nicer to me and facilitate the reading as well.

Comment (stsewd) [reply readthedocs/core/static-src/core/js/doc-embed/utils.js:6]: I'm also +1 on `:` instead of `-`. Checking if there is any style for that...

Comment (stsewd) [reply readthedocs/core/static-src/core/js/doc-embed/utils.js:6]: > If you provide a description, you can make the JSDoc comment more readable by inserting a hyphen before the description. Be sure to include a space before and after the hyphen.

From https://jsdoc.app/tags-param.html

Comment (ericholscher) [reply readthedocs/core/static-src/core/js/doc-embed/utils.js:6]: @stsewd seems this merged without this, just so you know.

Comment (stsewd) [reply readthedocs/core/static-src/core/js/doc-embed/utils.js:6]: Got it fixed in https://github.com/readthedocs/readthedocs.org/pull/9547

**Pull request merged**